ndonnx/_core/_impl.py

```diff
@@ -890,6 +890,8 @@ def shape(self, x):
         current = x
         while isinstance(current, ndx.Array):
             current = next(iter(current._fields.values()))
+        if current.ndim == 0:
+            current = ndx.reshape(current, [1])
```

Comment:
This looks fishy! A a scalar is not a rank-1 tensor with length 1. Furthermore, this fails under the following test:

```python
def test_shape():
    a = ndx.array(shape=(), dtype=ndx.int64)
    a.shape
```

Things pass if `a` is constant because the code paths are different.

Reply:
Indeed, this slipped in. Thanks for pointing it out.